x-pack/plugins/lens/public/shared_components/legend_action_popover.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiContextMenuPanelDescriptor, EuiIcon, EuiPopover, EuiContextMenu } from '@elastic/eui';
+import type { LensFilterEvent } from '../types';
+import { desanitizeFilterContext } from '../utils';
+
+export interface LegendActionPopoverProps {
+  /**
+   * Determines the panels label
+   */
+  label: string;
+  /**
+   * Callback on filter value
+   */
+  onFilter: (data: LensFilterEvent['data']) => void;
+  /**
+   * Determines the filter event data
+   */
+  context: LensFilterEvent['data'];
+}
+
+export const LegendActionPopover: React.FunctionComponent<LegendActionPopoverProps> = ({
+  label,
+  onFilter,
+  context,
+}) => {
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const panels: EuiContextMenuPanelDescriptor[] = [
+    {
+      id: 'main',
+      title: `${label}`,
```

Comment:
nitpick: why wrapping here?

Reply:
No reason at all, I missed that! Fixed https://github.com/elastic/kibana/pull/102026/commits/496c6601e3a7c7128ed8a877785f3dd802aac57a